provider/source/src/time_zones/convert.rs

```diff
@@ -96,9 +98,79 @@ fn compute_meta_zone_ids_btreemap(
     for (meta_zone_id, meta_zone_id_data) in meta_zone_ids_resource.iter() {
         meta_zone_ids.insert(meta_zone_id_data.long_id.to_string(), *meta_zone_id);
     }
+    // TODO(#1781): Remove this special case once the short id is updated in CLDR
+    meta_zone_ids.insert("Yukon".to_owned(), MetazoneId(tinystr!(4, "yuko")));
     meta_zone_ids
 }
 
+type Offsets = (EightsOfHourOffset, Option<EightsOfHourOffset>);
+
+// TODO: load this from a better source. Ideally this data would be in CLDR
+fn offsets_for_meta(meta_zones_territory: &Vec<MetazoneTerritory>, meta: &str) -> Option<Offsets> {
+    let (mut standard, mut daylight) = (None, None);
+    let mut first_zone = "";
+    for m in meta_zones_territory {
+        if m.map_zone.other != meta {
+            continue;
+        }
+        if m.map_zone.zone_type == "Europe/Minsk" {
+            // Minsk has an incorrect metazone, so ignore it
+            // https://unicode-org.atlassian.net/browse/CLDR-17927
+            continue;
+        }
+        let Some((new_standard, new_daylight)) = offsets_for_iana(&m.map_zone.zone_type) else {
+            continue;
+        };
+        if let Some(standard) = standard {
+            match (standard, daylight, new_standard, new_daylight) {
+                // Same standard times, no DST, allowed
+                (s1, None, s2, None) if s1 == s2 => {}
+                // Same standard times, one DST, allowed
+                (s1, Some(_), s2, None) | (s1, None, s2, Some(_)) if s1 == s2 => {}
+                // Same standard times, same DST times, allowed
+                (s1, Some(d1), s2, Some(d2)) if s1 == s2 && d1 == d2 => {}
+                // One tz has DST, and it's the other's standard time, sketchy but allowed
+                (s, None, _, Some(d)) | (_, Some(d), s, None) if s == d => {}
+                // Both tzs have DST but swapped, sketchy but allowed
+                (s1, Some(d1), s2, Some(d2)) if s1 == d2 && d1 == s2 => {}
+                // all other cases
+                _ => {
+                    log::warn!(
+                        "{:?} has different offsets from {first_zone:?}: standard {} vs {}, daylight {:?} vs {:?}",
+                        m.map_zone.zone_type,
+                        new_standard as f32 / 8.0,
+                        standard as f32 / 8.0,
+                        new_daylight.map(|o| o as f32 / 8.0),
+                        daylight.map(|o| o as f32 / 8.0),
+                    )
```

Comment:
Question: do you get any warnings printed?

Reply:
No, only the Minsk thing.

Reply:
These are the warnings when I don't allow DST divergence: 
```
"Africa/Accra" has different offsets from "Atlantic/Reykjavik": standard 0 vs 0, daylight None vs Some(1.0)
"Africa/Bamako" has different offsets from "Atlantic/Reykjavik": standard 0 vs 0, daylight None vs Some(1.0)
"Africa/Banjul" has different offsets from "Atlantic/Reykjavik": standard 0 vs 0, daylight None vs Some(1.0)
"Africa/Conakry" has different offsets from "Atlantic/Reykjavik": standard 0 vs 0, daylight None vs Some(1.0)
"Africa/Dakar" has different offsets from "Atlantic/Reykjavik": standard 0 vs 0, daylight None vs Some(1.0)
"Africa/Freetown" has different offsets from "Atlantic/Reykjavik": standard 0 vs 0, daylight None vs Some(1.0)
"Africa/Lome" has different offsets from "Atlantic/Reykjavik": standard 0 vs 0, daylight None vs Some(1.0)
"Africa/Nouakchott" has different offsets from "Atlantic/Reykjavik": standard 0 vs 0, daylight None vs Some(1.0)
"Africa/Tunis" has different offsets from "Europe/Paris": standard 1 vs 1, daylight None vs Some(2.0)
"America/Anguilla" has different offsets from "America/Halifax": standard -4 vs -4, daylight None vs Some(-3.0)
"America/Antigua" has different offsets from "America/Halifax": standard -4 vs -4, daylight None vs Some(-3.0)
"America/Aruba" has different offsets from "America/Halifax": standard -4 vs -4, daylight None vs Some(-3.0)
"America/Barbados" has different offsets from "America/Halifax": standard -4 vs -4, daylight None vs Some(-3.0)
"America/Belize" has different offsets from "America/Chicago": standard -6 vs -6, daylight None vs Some(-5.0)
"America/Cayman" has different offsets from "America/New_York": standard -5 vs -5, daylight None vs Some(-4.0)
"America/Costa_Rica" has different offsets from "America/Chicago": standard -6 vs -6, daylight None vs Some(-5.0)
"America/Curacao" has different offsets from "America/Halifax": standard -4 vs -4, daylight None vs Some(-3.0)
"America/Dominica" has different offsets from "America/Halifax": standard -4 vs -4, daylight None vs Some(-3.0)
"America/El_Salvador" has different offsets from "America/Chicago": standard -6 vs -6, daylight None vs Some(-5.0)
"America/Grenada" has different offsets from "America/Halifax": standard -4 vs -4, daylight None vs Some(-3.0)
"America/Guadeloupe" has different offsets from "America/Halifax": standard -4 vs -4, daylight None vs Some(-3.0)
"America/Guatemala" has different offsets from "America/Chicago": standard -6 vs -6, daylight None vs Some(-5.0)
"America/Jamaica" has different offsets from "America/New_York": standard -5 vs -5, daylight None vs Some(-4.0)
"America/Kralendijk" has different offsets from "America/Halifax": standard -4 vs -4, daylight None vs Some(-3.0)
"America/Lower_Princes" has different offsets from "America/Halifax": standard -4 vs -4, daylight None vs Some(-3.0)
"America/Marigot" has different offsets from "America/Halifax": standard -4 vs -4, daylight None vs Some(-3.0)
"America/Martinique" has different offsets from "America/Halifax": standard -4 vs -4, daylight None vs Some(-3.0)
"America/Mexico_City" has different offsets from "America/Chicago": standard -6 vs -6, daylight None vs Some(-5.0)
"America/Montserrat" has different offsets from "America/Halifax": standard -4 vs -4, daylight None vs Some(-3.0)
"America/Panama" has different offsets from "America/New_York": standard -5 vs -5, daylight None vs Some(-4.0)
"America/Port_of_Spain" has different offsets from "America/Halifax": standard -4 vs -4, daylight None vs Some(-3.0)
"America/Puerto_Rico" has different offsets from "America/Halifax": standard -4 vs -4, daylight None vs Some(-3.0)
"America/St_Kitts" has different offsets from "America/Halifax": standard -4 vs -4, daylight None vs Some(-3.0)
"America/St_Lucia" has different offsets from "America/Halifax": standard -4 vs -4, daylight None vs Some(-3.0)
"America/St_Thomas" has different offsets from "America/Halifax": standard -4 vs -4, daylight None vs Some(-3.0)
"America/St_Vincent" has different offsets from "America/Halifax": standard -4 vs -4, daylight None vs Some(-3.0)
"America/Tegucigalpa" has different offsets from "America/Chicago": standard -6 vs -6, daylight None vs Some(-5.0)
"America/Tortola" has different offsets from "America/Halifax": standard -4 vs -4, daylight None vs Some(-3.0)
"Atlantic/St_Helena" has different offsets from "Atlantic/Reykjavik": standard 0 vs 0, daylight None vs Some(1.0)
```

It appears that metazones don't imply DST equality.

Reply:
I think those error messages all are of the form "standard x vs x, daylight y vs None", yes? So that just means there are some countries sharing the same metazone but not all observe DST. It should still be safe to add the display name.